fw/tls.c

```diff
@@ -611,9 +611,11 @@ tfw_tls_conn_dtor(void *c)
 {
 	struct sk_buff *skb;
 	TlsCtx *tls = tfw_tls_context(c);
+	bool hs_was_done = false;
 
-	if (TFW_FSM_TYPE(((TfwConn *)c)->proto.type) == TFW_FSM_H2)
-		tfw_h2_context_clear(tfw_h2_context(c));
+	if (TFW_CONN_PROTO((TfwConn *)c) == TFW_FSM_H2
+	    && (hs_was_done = ttls_hs_done(tls)))
```

Comment:
Why do we need `hs_was_done`?

Reply:
Yes we can remove this variable. First of all i plan to pass it to `tfw_cli_conn_release`, to clear &ctx->skb_head only for intialized context. But I understand, that we cant do it in a good way. Better to clear &ctx->skb_head in context cleanup function